src/core/server/config/integration_tests/config_deprecation.test.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { mockLoggingService } from './config_deprecation.test.mocks';
+import { loggingServiceMock } from '../../logging/logging_service.mock';
+import * as kbnTestServer from '../../../../test_utils/kbn_server';
+
+describe('configuration deprecations', () => {
+  let root: ReturnType<typeof kbnTestServer.createRoot>;
+
+  afterEach(async () => {
+    if (root) {
+      await root.shutdown();
+    }
+  });
+
+  it('should not log deprecation warnings for default configuration', async () => {
+    root = kbnTestServer.createRoot();
+
+    await root.setup();
+
+    const logs = loggingServiceMock.collect(mockLoggingService);
+    expect(logs.warn).toMatchInlineSnapshot(`Array []`);
+  });
+
+  it('should log deprecation warnings for core deprecations', async () => {
+    root = kbnTestServer.createRoot({
+      optimize: {
+        lazy: true,
+        lazyPort: 9090,
+      },
+    });
+
+    await root.setup();
+
+    const logs = loggingServiceMock.collect(mockLoggingService);
+    expect(logs.warn).toMatchInlineSnapshot(`
+      Array [
+        Array [
+          "\\"optimize.lazy\\" is deprecated and has been replaced by \\"optimize.watch\\"",
```

Comment:
a disadvantage of such testing is lack of context testing. (`config.deprecation`)

Reply:
I agree. However I think I would prefer to improve the logger mock to allow to filter by context than actually use `child_process.spawn` is possible, as we may also want it in unit test (and I already do in this PR for the added test in `src/core/server/server.test.ts`). However this is a big change and I would prefer to do it in a separate PR.

But WDYT?

Reply:
In theory we could leverage new config format for NP logging https://github.com/elastic/kibana/issues/41956 to filter logs by context and write them into inMemoryAppender. That would require minimal boilerplate code.

Reply:
`BufferAppender` you mean?  
- It is a *reserved* appender that can't be set via configuration file
- It will still requires some mocking or trick to access the appender logs, as it's not exposed currently

Reply:
I didn't mean the concrete implementation. We can implement a custom appender for testing purposes only.

Reply:
The code in `src/core/server/logging/appenders/appenders.ts` make me think that this would mean ship test-only code, as there is currently no way to register custom appenders (and I don't think we planned/want that?), so we would still relies on some mocking mechanism (to mock `Appenders.create` for example), so not sure of the added value in comparison to put this logging into the logger mock?

But I think testing tools in integration tests (aka when using `kbnTestServer`) is a discussion on it's own.

Reply:
> and I don't think we planned/want that?

I have no clue to what extent we want to support log4j2 configuration. It was just an idea of testing the whole logging pipeline (including filtering & appenders).